### PR TITLE
Support openapi formats

### LINKED ISSUE #18: add basic formats
for example : int32, double timestamp

### PR DESCRIPTION
**Pull request by kobik**
Closes #18

### CODE REVIEW

File: src/validators/formatValidators.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    double: {
+        max: new Decimal(2).pow(1024),
+        min: new Decimal(2).pow(1024).negated()
+    },
+    byte: {
+        regex: /^[a-zA-Z0-9/+]+={0,2}$/
+    }
+};
+
+module.exports.defaultFormatsValidators = [
```
Comment (kibertoad) [inline src/validators/formatValidators.js]: It's considered a good practice to separate export definition from actual implementation, see Revealing Module Pattern: https://weblogs.asp.net/dwahlin/techniques-strategies-and-patterns-for-structuring-javascript-code-revealing-module-pattern

**Comment by kibertoad** [comment]
This is an amazing improvement, appreciate you spending time on it!

File: src/validators/formatValidators.js (hunk 1 above)
Comment (kobik) [reply src/validators/formatValidators.js]: Thanks @kibertoad, read your link about revealing pattern and I'm not sure that it fits the  need here, as all I do here is exporting a const.

Isn't changing the export to `module.exports = { defaultFormatsValidators  }` would be enough?

Comment (kibertoad) [reply src/validators/formatValidators.js]: > Isn't changing the export to module.exports = { defaultFormatsValidators } would be enough?

Yup, that's exactly what I'm suggesting :D. Sorry for the confusion.

**Pull request merged**